File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:

        mapp = {}

        for i in range(len(nums)):
            if nums[i] not in mapp:
                mapp[nums[i]] = i
            else:
                if abs(i - mapp[nums[i]]) <= k:
                    return True
                
                mapp[nums[i]] = i

        return False

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        set<int> vis;

        for(int i=0;i<nums.size();i++)
        {
            if(vis.count(nums[i]))
            {
                return true;
            }

            vis.insert(nums[i]);

            if(vis.size() > k)
            {
                vis.erase(nums[i-k]);
            }
        }

        return false;
    }
};